src/custom/hooks/useApproveCallback/useApproveCallbackMod.ts

```diff
@@ -44,13 +53,13 @@ export function useApproveCallback(
     if (!currentAllowance) return ApprovalState.UNKNOWN
 
     // Return approval state
-    if (currentAllowance.lessThan(amountToApprove)) {
+    if ((amountToCheckAgainstAllowance || currentAllowance).lessThan(amountToApprove)) {
```

Comment:
Does this work?
Shouldn't the `lessThan` be applied individually and then do the `OR`?

Also, if it is like we discussed, `amountToCheckAgainstAllowance` is always be < `amountToApprove` because the later will always be MAX_UINT.
Thus you should compare `amountToCheckAgainstAllowance` only against `currentAllowance`
Or did I get this wrong?

Reply:
I think u are right.

For example, for investment 1000 USDC in CoW Protocol:
- amountToCheckAgainstAllowance: 1000 USDC
- amountToApprove: MAX_UINT

if currentAllowance >= amountToCheckAgainstAllowance ---> all good
else set allowance to MAX_UINT

Reply:
@anxolin @alfetopito 6b2ba55